### PR TITLE
CC-32 # Added bucketPathPrefix option

### PR DESCRIPTION
**Pull request by mymattcarroll**
### Added

-   CC-32: `bucketPathPrefix` option to allow uploading to a prefixed path in S3 buckets

### CODE REVIEW

File: lib/utils/bucket-path-prefix-helper.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+function validate (
+  bucketPathPrefix /* : string | void */
+) /* : string */ {
+  if (!bucketPathPrefix) {
+    return '';
+  }
+
+  // prefix should have a slack at the end or be an empty string.
+  if (!bucketPathPrefix.endsWith('/')) {
```
Comment (simonmarklar) [inline lib/utils/bucket-path-prefix-helper.js]: this requires node 6.10.3 (which is the current LTS, about 2 weeks old), but `package.json` says it needs node 4. Im happy using the function if we bump the `package.json` version

also its called a "slash" not a "slack" :P

File: test/upload.js [hunk 2]
```diff
@@ -122,7 +122,7 @@ test('task uploads and deletes', (t) => {
     'ijkl.txt': {deleting: false, deleted: false}
   };
   const cwd = path.join(__dirname, 'fixtures');
-  const task = upload({cwd, s3: mockS3, prune: true});
+  const task = upload({cwd, s3: mockS3, prune: true, bucketPathPrefix: '//dev'});
```
Comment (simonmarklar) [inline test/upload.js]: did you mean to have the `//` here ?

Comment (mymattcarroll) [inline test/upload.js]: yeah its part of the tests to ensure that both of them get removed.

File: lib/utils/bucket-path-prefix-helper.js (hunk 1 above)
Comment (mymattcarroll) [inline lib/utils/bucket-path-prefix-helper.js]: Apparently is also available on Node 4.8.3 :)

Fixed typo.

**Pull request merged**